flycheck-rust.el

```diff
@@ -103,7 +135,8 @@ Flycheck according to the Cargo project layout."
   (interactive)
   (when (buffer-file-name)
     (-when-let (root (flycheck-rust-project-root))
-      (let ((rel-name (file-relative-name (buffer-file-name) root)))
+      (let ((rel-name (file-relative-name (buffer-file-name) root))
+            (target (flycheck-rust-find-target (buffer-file-name))))
```

Comment:
As we don't need `target` itself but only its components, please use `pcase-let` instead of `let` to destructure the pair immediately.  I think the binding form would be ``(`(,target-type . ,target-name) (flycheck-rust-find-target …))``.

Makes no difference semantically, but it's much easier to read :)